### ISSUE TITLE
Unable to pull image from docker registry 

### ISSUE THREAD

**Issue by yadneshk** [virt-manager/virt-bootstrap#1]
I am using Fedora27 and have installed all the required dependencies.
Trying to pull docker image and extract it to obtain rootfs and then provide it to virt-install to run that container on LXC.
I am getting the below error 
[root@client virt-bootstrap]# ./run src/virtBootstrap/virt_bootstrap.py docker://docker.io/fedora /tmp/newfed
INFO    : Checking cached layers
INFO    : Downloading container image
Traceback (most recent call last):
  File "src/virtBootstrap/virt_bootstrap.py", line 347, in <module>
    sys.exit(main())
  File "src/virtBootstrap/virt_bootstrap.py", line 335, in main
    progress_cb=args.status_only)
  File "src/virtBootstrap/virt_bootstrap.py", line 149, in bootstrap
    progress=prog).unpack(dest)
  File "/root/virt-bootstrap/src/virtBootstrap/sources/docker_source.py", line 279, in unpack
    self.fetch_layers()
  File "/root/virt-bootstrap/src/virtBootstrap/sources/docker_source.py", line 267, in fetch_layers
    self.download_image()
  File "/root/virt-bootstrap/src/virtBootstrap/sources/docker_source.py", line 152, in download_image
    self.read_skopeo_progress(skopeo_copy)
  File "/root/virt-bootstrap/src/virtBootstrap/sources/docker_source.py", line 243, in read_skopeo_progress
    raise subprocess.CalledProcessError(proc.returncode, ' '.join(cmd))
subprocess.CalledProcessError: Command 'skopeo copy docker://docker.io/fedora dir:/var/cache/virt-bootstrap/docker_images' returned non-zero exit status 1

**Comment by rst0git** [comment]
Hi @yadneshk,

Try with `--no-cache` flag. For example:

```
 ./run src/virtBootstrap/virt_bootstrap.py docker://docker.io/fedora /tmp/newfed --no-cache
```
If this works then cleaning up previously downloaded images should resolve your problem:

```
rm /var/cache/virt-bootstrap/docker_images/*
```

Otherwise what version of skopeo and what is the output of
```
skopeo copy docker://docker.io/fedora dir:/var/cache/virt-bootstrap/docker_images
```

**Comment by khomesh24** [comment]
Hi @rst0git, 
I am working with @yadneshk. 
I tried with ``` --no-cache ``` flag but I am getting same error. I have also cleaned ``` /var/cache/virt-bootstrap/docker_images/  ``` directory. 

skopeo version I am using is  
```skopeo version 0.1.26 commit: 25a4b84ef5282cd87116dee32260a2ea16a58104```

and output of 
``` skopeo copy docker://docker.io/fedora dir:/var/cache/virt-bootstrap/docker_images ``` 
is 
``` FATA[0000] Error initializing destination dir:/var/cache/virt-bootstrap/docker_images: not a containers image directory, don't want to overwrite important data ```

**Comment by rst0git** [comment]
Hi @khomesh24 and @yadneshk,

The error message below occurs because `skopeo copy` has changed its behaviour with [this commit](containers/image/commit/aa0676d506667ff132e7a8b0c464d531b84c2640).

```
FATA[0000] Error initializing destination dir:/var/cache/virt-bootstrap/docker_images: not a containers image directory, don't want to overwrite important data
```
The commit aim is for skopeo to store only a single container image per directory. After successful image download is created a file called `version` in the destination directory.
When `skopeo copy` is called it checks if the destination directory is empty or if the `version` file exist and have correct content.
If the file does not exist (or have incorrect content) and the destination folder is not empty it will raise the above error.
For more details see [lines](https://github.com/containers/image/blob/f950aa3529148eb0dea90888c24b6682da641b13/directory/directory_dest.go#L43-L71).

I send [a patch](https://www.redhat.com/archives/virt-tools-list/2017-December/msg00004.html) for virt-bootstrap which aims to resolve this issue. I will wait for the patch to be reviewed before merging it with master.

**Comment by rst0git** [comment]
Fixed with commit [76a945f9f2b88407bac8f7a8cc0d1464db183f21](https://github.com/virt-manager/virt-bootstrap/commit/76a945f9f2b88407bac8f7a8cc0d1464db183f21)
